### PR TITLE
[RFC] terminal escape codes/declarations of colorama

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: src/_pytest/_io/terminalwriter.py [hunk 1]
```diff
@@ -37,32 +41,39 @@ def should_do_markup(file: TextIO) -> bool:
...
-        invert=7,
-    )
+    hasmarkup: bool
+    code_highlight: bool
 
-    def __init__(self, file: Optional[TextIO] = None) -> None:
+    def __init__(
+        self,
+        file: Optional[TextIO] = None,
+        dup=False,
```
Comment (nicoddemus) [inline src/_pytest/_io/terminalwriter.py:73]:
```suggestion
        dup: bool=False,
```

Actually it doesn't appear to be used?

File: src/_pytest/_io/terminalwriter.py [hunk 2]
```diff
@@ -37,32 +41,39 @@ def should_do_markup(file: TextIO) -> bool:
...
+        has_markup: Optional[bool] = None,
+        code_highlight: bool = True,
+    ) -> None:
         if file is None:
             file = sys.stdout
         if hasattr(file, "isatty") and file.isatty() and sys.platform == "win32":
```
Comment (nicoddemus) [inline src/_pytest/_io/terminalwriter.py:79]: We can now remove the `try/except` below

File: src/_pytest/_io/terminalwriter.py [hunk 3]
```diff
@@ -37,32 +41,39 @@ def should_do_markup(file: TextIO) -> bool:
...
-        blink=5,
-        invert=7,
-    )
+    hasmarkup: bool
+    code_highlight: bool
 
-    def __init__(self, file: Optional[TextIO] = None) -> None:
+    def __init__(
+        self,
+        file: Optional[TextIO] = None,
```
Comment (nicoddemus) [inline src/_pytest/_io/terminalwriter.py:72]:
```suggestion
        file: Optional[TextIO] = None,
        *,
```

File: src/_pytest/_io/terminalwriter.py [hunk 4]
```diff
@@ -37,32 +41,39 @@ def should_do_markup(file: TextIO) -> bool:
     )
 
 
+def _ansi_items(inst, transform: Callable[[str], str] = str.lower) -> Dict[str, str]:
+    return {transform(name): cast(str, item) for name, item in vars(inst).items()}
+
+
+ESC_TABLE = {
```
Comment (nicoddemus) [inline src/_pytest/_io/terminalwriter.py:48]: I'm curious, "esc" comes from "escape"?

**Review by nicoddemus — APPROVED** [review]
Looks good, left a few suggestions.

**Comment by bluetech** [comment]
What's the rationale/reason for these changes?

File: src/_pytest/_io/terminalwriter.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/_pytest/_io/terminalwriter.py:73]: @nicoddemus whops, i  intend to add a dup, as we need to have a extra safe fd for printing when capture is active, i should have stashed it (or added it)

**Comment by RonnyPfannschmidt** [comment]
@bluetech just trying to move own hardcoded tables out of pytest and as colorama did most of the work (+- some name changes) its a good tryout

im hoping to do some experimentation around using rich for more details (as a plugin)
before that i want to move some code out of the terminal-writer and ensure good handling of some details

**Comment by bluetech** [comment]
> just trying to move own hardcoded tables out of pytest and as colorama did most of the work (+- some name changes) its a good tryout

IMO, it is not worth it to add a dependency for the few constant names.

Since rich depends on colorama, it will definitely make sense if/when we depend on rich, but until then I think it's better to avoid.

**Comment by nicoddemus** [comment]
> IMO, it is not worth it to add a dependency for the few constant names.

FTR I don't mind the extra dependency (on Linux, it was already a hard dependency on Windows).

**Comment by RonnyPfannschmidt** [comment]
for completion fs dup hnadling hasto be done (so the fd for the main terminal can be something else

(pytest-print breaks due to terminalwriter no longer using the capture dup fd)

**Comment by RonnyPfannschmidt** [comment]
closing as abadonded

**Pull request closed without merging**